src/utils/kmlUtils.js

```diff
@@ -488,4 +489,65 @@ export function parseKml(kmlLayer, projection, iconSets) {
     return features
 }
 
+/**
+ * The goal of this function is to transform internal KML's URLs into service-proxy equivalents.
+ * This is done for backward compatibility, and is heavily influcend/copied from mf-geoadmin3 code
+ * found here :
+ * https://github.com/geoadmin/mf-geoadmin3/blob/6a7b99a2cc9980eec27b394ee709305a239549f1/src/components/map/KmlService.js#L60-L106
+ *
+ * @param kmlData
+ * @returns {String} The kmlData, with URLs/HREFs transformed if needed
+ */
+export function proxifyKmlData(kmlData) {
+    // Replace space-only ids
+    let transformedKml = kmlData.replace(/id=(["'])\s*(["'])/g, '')
+
+    // Add ids for the link with 3d object
+    let id = Date.now()
+    transformedKml = transformedKml.replace(/<Placemark\s*>/g, function () {
+        return `<Placemark id="${id++}">`
+    })
+
+    // Replace all hrefs to prevent errors if image doesn't have
+    // CORS headers. Exception for *.geo.admin.ch, *.bgdi.ch and google
+    // markers icons (only https)
+    // to keep the OL magic for anchor origin.
+    // Test regex here: http://regex101.com/r/tF3vM0/9
+    // List of Google icons: http://www.lass.it/Web/viewer.aspx?id=4
+    transformedKml = transformedKml.replace(
+        /<href>http(?!(s:\/\/maps\.(google|gstatic)\.com[a-zA-Z\d.\-/_]*\.png|s?:\/\/[a-z\d.-]*(bgdi|geo.admin)\.ch))/g,
```

Comment:
In the old viewer, in the first `[a-zA-Z\d.\-/_]` , both the `.` and the `/` are also escaped.
in the second one `[a-z\d.-]`, the `.` and the `-` are escaped. 
There is no need to escape them in js, but I would still change the documentation line regarding the regex test permalink to reflect the current regex